optimum/exporters/openvino/model_configs.py

```diff
@@ -4319,19 +4320,29 @@ def __init__(
 
         config = normalized_config.config
         self.intermediate_size = int(config.mamba_expand * config.hidden_size)
-        self.ssm_state_size = config.mamba_d_state
         self.conv_kernel_size = config.mamba_d_conv
-        self.n_mamba_heads = config.n_mamba_heads
-        self.mamba_ngroups = config.mamba_ngroups
         self.mamba_d_state = config.mamba_d_state
-        self.mamba_headdim = config.mamba_headdim
-        self.head_dim = config.attention_head_dim
-        self.hybrid_layer_ids = config.hybrid_layer_ids
-        logger.warning(
-            "The current support for the 'Zamba2' model type is experimental. "
-            "Performance is not optimal with high memory consumption. "
-            "Optimizations and improved support will be available in a future OpenVINO release."
-        )
+        if config.model_type == "zamba2":
+            self.n_mamba_heads = config.n_mamba_heads
+            self.mamba_ngroups = config.mamba_ngroups
+            self.mamba_headdim = config.mamba_headdim
+            self.head_dim = config.attention_head_dim
+            self.num_hybrid_layers = len(config.hybrid_layer_ids)
+            logger.warning(
+                "The current support for the 'Zamba2' model type is experimental. "
+                "Performance is not optimal with high memory consumption. "
+                "Optimizations and improved support will be available in a future OpenVINO release."
+            )
+        else:
+            # currently, this else-branch is applied for GraniteMoeHybrid models
+            self.n_mamba_heads = config.mamba_n_heads
+            self.mamba_ngroups = config.mamba_n_groups
+            self.mamba_headdim = config.mamba_d_head
+            self.head_dim = getattr(config, "head_dim", config.hidden_size // config.num_attention_heads)
+            self.num_hybrid_layers = config.layer_types.count("attention")
+            self.num_layers = config.layer_types.count("mamba")
```

Comment:
here `self.num_hybrid_layers` + `self.num_layers` is equal to `config.num_hidden_layers` is this correct ? I think we should rename fo clarity (num_layers here is actually the number of mamba layers), or at least add a comment to explain